### PR TITLE
feat: fetching pwa settings

### PR DESCRIPTION
**Pull request by clincu-plenty**
## Why:

Closes: #ID

## Checklist before requesting a review

- [ ] My code follows the code style of this project.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] I've updated `docs/changelog/changelog_en.md` and `docs/changelog/changelog_de.md`. If I'm a non-German speaker, I've still updated the file with the English version as a placeholder.

### CODE REVIEW

File: apps/web/.eslintignore [hunk 1]
```diff
@@ -1,3 +1,5 @@
 nuxt.config.js
 jest.setup.ts
 cypress.config.ts
+nuxt.config.ts
+utils/configurationHelper.ts
```
Comment (maxiroellplenty) [inline apps/web/.eslintignore]: please remove and disable specific rules if necessary.

File: apps/web/utils/configurationHelper.ts [hunk 2]
```diff
@@ -0,0 +1,109 @@
+import axios from 'axios';
+import https from 'https';
+import fs from 'fs';
+import os from 'os';
+import path from 'path';
+
+const envFilePath = path.resolve(__dirname, "../.env");
+const envTmpFilePath = path.resolve(__dirname, '../.env.tmp');
+
+// read .env file & convert to array
```
Comment (maxiroellplenty) [inline apps/web/utils/configurationHelper.ts]: We usually don't write comments in the app, if the code is complex we move it into a function and explain what the code does by having a descriptive function name.

File: apps/web/utils/configurationHelper.ts [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    // For last iteration just skip adding
+    if (i !== requiredEnvVars.length - 1) {
+      requiredEnvData += '\n';
+    }
+  }
+
+  console.log('Required env variables found ✅');
+
+  // Create .env.tmp file
+  console.log('Creating .env.tmp file...');
```
Comment (maxiroellplenty) [inline apps/web/utils/configurationHelper.ts]: I would only have one console log that confirms writing the env file.
And only log errors and use some tryCatch around the operations.

**Review by maxiroellplenty — CHANGES_REQUESTED** [review]

**Review by ksted — DISMISSED** [review]

**Review by maxiroellplenty — DISMISSED** [review]

File: apps/web/build/fetchConfiguration.ts [hunk 4]
```diff
@@ -72,13 +73,13 @@ const convertTemporaryToPermanentEnvironment = () => {
...
 const fetchConfiguration = async () => {
-  if (environmentMap.DISABLE_SETTINGS_FETCH === 'undefined' || environmentMap.DISABLE_SETTINGS_FETCH === '0') {
+  if (environmentMap.DISABLE_SETTINGS_FETCH === undefined || environmentMap.DISABLE_SETTINGS_FETCH === '1') {
     console.warn(`Fetching PWA settings is disabled! Check DISABLE_SETTINGS_FETCH in .env file.`);
     return;
   }
 
   setupTemporaryEnvironment();
-  fetchAndWriteRemoteConfiguration();
+  await fetchAndWriteRemoteConfiguration();
```
Comment (clincu-plenty) [inline apps/web/build/fetchConfiguration.ts:82]: Lint threw error. Had to fix it.

File: apps/web/build/fetchConfiguration.ts [hunk 5]
```diff
@@ -72,13 +73,13 @@ const convertTemporaryToPermanentEnvironment = () => {
 };
 
 const fetchConfiguration = async () => {
-  if (environmentMap.DISABLE_SETTINGS_FETCH === 'undefined' || environmentMap.DISABLE_SETTINGS_FETCH === '0') {
+  if (environmentMap.DISABLE_SETTINGS_FETCH === undefined || environmentMap.DISABLE_SETTINGS_FETCH === '1') {
```
Comment (clincu-plenty) [inline apps/web/build/fetchConfiguration.ts:76]: Fixing both cases, first case datatype is undefined not 'undefined' as string.
Second case, when the DISABLE_SETTINGS_FETCH is true => No fetching, so we throw the warn.

File: apps/web/build/fetchConfiguration.ts [hunk 6]
```diff
@@ -18,8 +18,9 @@ const setupTemporaryEnvironment = () => {
   let requiredEnvironmentData = '';
 
   for (const [key, value] of Object.entries(environmentMap)) {
-    if (key !== 'DISABLE_SETTINGS_FETCH' && !value) {
-      console.error(`Missing or invalid required environment variable: ${value}`);
+    if (key === 'DISABLE_SETTINGS_FETCH') continue;
```
Comment (clincu-plenty) [inline apps/web/build/fetchConfiguration.ts:21]: I think this might be a better solution? Looks better from my POV.

**Review by ksted — APPROVED** [review]

**Pull request merged**